source/git.js

```diff
@@ -353,7 +356,7 @@ git.commit = function(repoPath, amend, message, files) {
   if (message === undefined)
     return task.setResult({ error: 'Must specify commit message' });
 
-  if ((!(files instanceof Array) || files.length == 0) && !amend)
+  if ((!(Object.prototype.toString.call(files) === '[object Array]') || files.length == 0) && !amend)
```

Comment:
Hi,
I used the commit API, and it rejected my commit because of the following test in "git.js" :  if ((!(files instanceof Array) || files.length == 0) && !amend)    return task.setResult({ error: 'Must specify files or amend to commit' });It turns out that the test "instanceof Array" fails.As per some blogs (e.g. http://perfectionkills.com/instanceof-considered-harmful-or-how-to-write-a-robust-isarray/),http://perfectionkills.com/instanceof-considered-harmful-or-how-to-write-a-robust-isarray/), the test should be written :I made the suggested modification, and everything worked like a charm :)
Kind regards,Olivier LE QUELLEC
Date: Thu, 23 Jul 2015 06:16:48 -0700
From: notifications@github.com
To: ungit@noreply.github.com
CC: olivier.le-quellec@live.fr
Subject: Re: [ungit] Catch git spawn exception / better test for Array type (#598)

In source/git.js:

> @@ -353,7 +356,7 @@ git.commit = function(repoPath, amend, message, files) {
>    if (message === undefined)
>      return task.setResult({ error: 'Must specify commit message' });
>  
> -  if ((!(files instanceof Array) || files.length == 0) && !amend)
> +  if ((!(Object.prototype.toString.call(files) === '[object Array]') || files.length == 0) && !amend)

Just curious: how is this better? Isn't instanceof the correct way of checking types in JavaScript?


—
Reply to this email directly or view it on GitHub.